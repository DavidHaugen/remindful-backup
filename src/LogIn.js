import React, { Component } from 'react';
import { Link } from 'react-router-dom'

class LogIn extends Component {
  render(){
    return(
      <div>
        <h2>Welcome back!</h2>
        <h3>Please log in below:</h3>
        <form>
          <div>
            <label>Email address
              <input />
            </label>
          </div>
          <div>
            <label>Password
              <input type='password'/>
            </label>
          </div>
          <button type='submit'>Submit</button>
        </form>
    <p>Not signed up yet? {<Link className='nav-link' to='/sign-up'> Sign up instead!</Link>}</p>
      </div>
    )
  }
}

export default LogIn