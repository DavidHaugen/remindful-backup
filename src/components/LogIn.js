import React, { Component } from 'react';
import { Link } from 'react-router-dom'
import AuthApiService from '../services/auth-api-service'
import TokenService from '../services/token-service'
import GoalsContext from '../context/GoalsContext'

class LogIn extends Component {
  static defaultProps = {
    onLoginSuccess: () => {    }
  }
  static contextType = GoalsContext

  state = { 
    error: null}

  handleSubmitJwtAuth = ev => {
    ev.preventDefault()
    this.setState({ error: null })
    this.context.loadingTrue();
    const { email_address, password } = ev.target
 
    AuthApiService.postLogin({
      email_address: email_address.value,
      password: password.value,
    })
      .then(res => {
        email_address.value = ''
        password.value = ''
        TokenService.saveAuthToken(res.authToken)
        this.props.history.push('/my-goals')
        this.context.getGoals()
        this.context.loadingFalse()
      })
      .catch(res => {
        this.setState({ error: res.error})
        this.context.loadingFalse()
      })
  }

  render(){
    const { error } = this.state

    if(this.context.loading){
      return(
        <div className="main" role="main">
          <div className="lds-grid"><div></div><div></div><div></div><div></div><div></div><div></div><div></div><div></div><div></div></div>
        </div>
      )
    } else{
    return(
      <div className="main" role="main">
        <div className="wrapper">
          <h1 className="karla">Welcome back!</h1>
          <p className="instructions">Please log in below:</p>
        <div className="formContainer">
          <form onSubmit={this.handleSubmitJwtAuth} className="inputForm">
            <div role='alert'>
              {error && <p className='red'>{error}</p>}
            </div>
            <div className='formField'>
              <label htmlFor="login_email_address" className="inputLabel">Email</label>
                <input name='email_address' type='email' id='login_email_address' className="inputField" required/>
            </div>
            <div className='formField'>
              <label htmlFor="password" className="inputLabel">Password</label>
                <input name='password' type='password' id='password' className="inputField" required/>
            </div>
            <button type='submit' className="textButton login">Log in</button>
          </form>
        </div>
        <p>Not signed up yet? {<Link className='signUpLink' to='/sign-up'>Sign up instead!</Link>}</p>
        </div>
      </div>
    )
  }
}
}

export default LogIn