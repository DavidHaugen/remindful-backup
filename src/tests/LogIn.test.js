import React from 'react';
import {MemoryRouter} from 'react-router-dom'
import LogIn from '../components/LogIn'
import {mount} from 'enzyme';
import renderer from 'react-test-renderer'

describe('<LogIn />', () => {
  it('Renders without crashing', () => {
      mount(<MemoryRouter><LogIn/></MemoryRouter>);
  });
  it('renders the UI as expected', () => {
    const tree = renderer
      .create(<MemoryRouter><LogIn/></MemoryRouter>)
      .toJSON();
    expect(tree).toMatchSnapshot();  
  });
});