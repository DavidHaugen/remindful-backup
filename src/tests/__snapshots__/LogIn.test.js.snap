// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LogIn /> renders the UI as expected 1`] = `
<div
  className="main"
  role="main"
>
  <div
    className="wrapper"
  >
    <h1
      className="karla"
    >
      Welcome back!
    </h1>
    <p
      className="instructions"
    >
      Please log in below:
    </p>
    <div
      className="formContainer"
    >
      <form
        className="inputForm"
        onSubmit={[Function]}
      >
        <div
          role="alert"
        />
        <div
          className="formField"
        >
          <label
            className="inputLabel"
            htmlFor="login_email_address"
          >
            Email
          </label>
          <input
            className="inputField"
            id="login_email_address"
            name="email_address"
            onChange={[Function]}
            required={true}
            type="email"
            value=""
          />
        </div>
        <div
          className="formField"
        >
          <label
            className="inputLabel"
            htmlFor="password"
          >
            Password
          </label>
          <input
            className="inputField"
            id="password"
            name="password"
            onChange={[Function]}
            required={true}
            type="password"
            value=""
          />
        </div>
        <button
          className="textButton login"
          type="submit"
        >
          Log in
        </button>
      </form>
    </div>
    <p>
      Not signed up yet? 
      <a
        className="signUpLink"
        href="/sign-up"
        onClick={[Function]}
      >
        Sign up instead!
      </a>
    </p>
  </div>
</div>
`;
